src/main/java/com/purbon/kafka/topology/backend/FileBackend.java

```diff
@@ -50,13 +50,11 @@ public void createOrOpen() {
   @Override
   public void createOrOpen(Mode mode) {
     try {
-      writer = new RandomAccessFile(STATE_FILE_NAME, "rw");
-      Path path = Paths.get(STATE_FILE_NAME);
-      if (path.toFile().exists()) {
-        writer.seek(0);
-        if (mode.equals(Mode.TRUNCATE)) {
-          writer.getChannel().truncate(0);
-        }
+      if (this.writer != null) writer.close();
+      if (mode.equals(Mode.TRUNCATE)) {
+        this.writer = new FileWriter(STATE_FILE_NAME, false);
```

Comment:
what do you think about having something like:

```java
 this.writer = new FileWriter(STATE_FILE_NAME, !mode.equals(Mode.TRUNCATE));
```

Reply:
fine for me :)

Reply:
I will just turn it around so it is null-safe, enum first